### ISSUE TITLE
Duration of kube_pod_container_status_terminated_reason metrics

### ISSUE THREAD

**Issue by utsav2307** [kubernetes/kube-state-metrics#344]
I understand that this metrics is keeping track of pods which were terminated and the reason for it, but the metrics store data of how many days? Like the data will be of last two days or last 12 hours?

**Comment by brancz** [comment]
Prometheus metrics always only reflect the current state. In this case, it reflects whatever the Pod object says, that the Kubernetes API returns. The value for the `reason` label are `OOMKilled|Error|Completed|ContainerCannotRun`. Note there is always a metric for _all_ of those reasons, but only the metric that has the value `1`, is the current state.

**Comment by mheggeseth** [comment]
I find it really unlikely for a Prom scrape will catch a pod container while its current state is `terminated`. It's much more likely that `lastState` has `terminated` while the current `state` is either `running` or `waiting`.

**Comment by jonaz** [comment]
I agree here. Trying to make an alert when pods are OOMKilled. But its only in that state for milliseconds so `kube_pod_container_status_terminated_reason{reason="OOMKilled"} > 0`does not work. Is there a was i can check lastState instead?

**Comment by brancz** [comment]
Yeah we can absolutely add a metric that exposes the Last state.

**Comment by jonaz** [comment]
Anyone working on that? We need it to send alert to the correct team when their app OOMs in production... 

I could give it a try if no one has planned to add lastState.

**Comment by andredantasrocha** [comment]
Any updates on that? I am trying to implement a very similar alert...

**Comment by benoneb** [comment]
hi guys, no one got this solved?

**Comment by ivelichkovich** [comment]
You can turn it into a time series and use sum_over_time. To catch if anything has been OOMKilled in the last hour for example:
sum_over_time(kube_pod_container_status_terminated_reason{reason="OOMKilled",namespace="core-teams-dev"}[1h])

**Comment by akram** [comment]
+1

**Comment by dingobaby** [comment]
+1
Also very interested in being able to find the last state reason.  Additionally is there is reason why all the kube_pod_container_status_* metrics dont include the  node label?  Being able to track if particular nodes have higher than average pod failures is also something Im interested in.

**Comment by 3h4x** [comment]
I would love to monitor OOM with `kube-state-metrics` but in current state it's impossible.
🙏

**Comment by svitlanacs** [comment]
I would like to add one more vote on this thread.

Here are the tasks I'd like to achieve:
(1) alert whenever a certain pod goes into error state (either via waiting_reason or terminated_reason), 
(2) build out timestamp-based historical view of what was happening with a particular pod:
Container Creating -> ErrImagePull -> Running -> Terminating -> Running

While last state will help with task #1, is there anything we could do about enabling task #2 via kube-state-metrics as well?

**Comment by mxinden** [comment]
I guess one could introduce a `kube_pod_container_status_last_terminated_reason` metric. It seems like @brancz is fine accepting a pull request for this.

For anyone wanting to tackle this, I am happy to help. Changes would need to touch [`pod.go`](https://github.com/kubernetes/kube-state-metrics/blob/master/pkg/collectors/pod.go). `cs.LastTerminationState` should give you everything you need.

**Comment by benoittgt** [comment]
Hello

I am curious how do you monitor this metric. I use `changes` at the moment but it doesn't properly report first seen terminated reason.

```
sum by(container, reason) (changes(kube_pod_container_status_last_terminated_reason{container="xxxx"}[$__rate_interval])) > 0 
```

![Screenshot_2022-06-08_at_12_13_08](https://user-images.githubusercontent.com/8417720/172601580-62eaa68d-3d80-48f2-a000-9dc8d861cc6b.jpg)

**Comment by CharlieC3** [comment]
@benoittgt I was able to get your query to work as expected by just removing the `> 0` at the end:
`sum by(container, reason) (changes(kube_pod_container_status_last_terminated_reason{container="xxxx"}[$__rate_interval]))`
